src/html5/parser/document.rs

```diff
@@ -898,10 +894,54 @@ impl DocumentBuilder {
     }
 }
 
+/// Constructs an iterator from a given DocumentHandle.
+/// Note that this iterator is a snapshot of the DocumentHandle
+/// at the time of creation; it does NOT update with tree mutations
+/// and a new iterator must be used.
```

Comment:
now that I think about it, this may actually be wrong. We are just cloning the reference to the document so technically mutations would be reflected, the iterator would likely just be unreliable depending on what nodes have been inserted/mutated

Reply:
Nvm, after actually testing this, the iterator is "consumed" if calling next() until the last element and will return None even if appending more nodes to the doc.